Log.h

```diff
@@ -1,5 +1,6 @@
 #include <fstream>
 #include "OPUNetTransportLayer.h"
+#include <winsock2.h>

```

Comment:
This change made me curious.

I assume this was done because of the `GUID&` parameter to the `GumpGuid` method. Indeed, that type should be declared before use, though it is used as a reference, so a forward declaration should be fine.

From another perspective, this file includes "OPUNetTransportLayer.h", which uses `GUID` in a public struct, so it would need to define or include it from somewhere. A better question though, is why does Log.h include OPUNetTransportLayer.h? I'm thinking that dependency should be removed, and forward declares used when possible.

Looking more closely at this file, it tries to forward declare `ofstream`, but without the `std::` prefix (which is wrong), and then includes `fstream` anyway. I'm not sure it's really standards conformant to open up the `std` namespace, and forward declare something in it, so we're probably better off just using the include here. Though if you can find some sources to show me I'm wrong, I'd love to hear it. I'm also uncertain if a forward declare is sufficient to define the external variable.

Reply:
Wasn't too worried about the extern use of fstream. I'd like to remove fstream in future branches anyways.

Forward declares are fine. I was just fixing the lack of either a forward declare or include.